Added setup instructions for Linux (#93)
avantifellows/quiz-backend: README.md

### Local DB Setup

#### Linux Systems

> The following steps are for a Linux systems.

Reviewer: ```suggestion
The following steps are for a Linux system.
```
Grammar error

### Local DB Setup

> - The following steps are for a Linux.

Reviewer: for a linux what?

Also, would suggest to not keep this section heading as a bullet point, but a `h4` heading.
Like `#### Linux Systems`

And same for mac section
`### Mac Systems`

- To run the backend locally, you would need to setup a local instance of mongodb. The offical instructions [here](https://www.mongodb.com/docs/manual/tutorial/install-mongodb-on-ubuntu/) are good and simple enough to follow. Those steps are also listed down below.

  - Import the public key used by the package management system. From a terminal, install gnupg if it is not already available:
    ```
    sudo apt-get install gnupg

>     ```

Reviewer: here you have written two things. one is importing the public key and then you've written to install GnuPG.
But you've only added the command for GnuPG installation. What about the public key?

  - Create a list file for MongoDB.

>     Create the list file /etc/apt/sources.list.d/mongodb-org-6.0.list for your version of Ubuntu.To check the Ubuntu version the host is running, open a terminal or shell on the host and execute

Reviewer: ```suggestion
    Create the list file `/etc/apt/sources.list.d/mongodb-org-6.0.list` for your version of Ubuntu. To check the Ubuntu version the host is running, open a terminal or shell on the host and execute
```

### Local DB Setup

> #### Linux Systems

Reviewer 2: Can you make a table of contents list at the top of readme file? I think we have enough sections / subsections. There are TOC generators online just use them instead of linking each section manually.

  - Import the public key used by the package management system.
  Issue the following command to import the MongoDB public GPG Key from [here](https://pgp.mongodb.com/server-6.0.asc):
    ```
    curl -fsSL https://pgp.mongodb.com/server-6.0.asc | \
    sudo gpg -o /usr/share/keyrings/mongodb-server-6.0.gpg \
    --dearmor
    ```

  - From a terminal, install gnupg if it is not already available:
    ```
    sudo apt-get install gnupg
    ```
  - Create a list file for MongoDB.
    Create the list file `/etc/apt/sources.list.d/mongodb-org-6.0.list` for your version of Ubuntu. To check the Ubuntu version the host is running, open a terminal or shell on the host and execute
    ```
    lsb_release -dc
    ```

    According to the version, run the respective code on the terminal.

      - Ubuntu 22.04 (Jammy)
        ```
        echo "deb [ arch=amd64,arm64 signed-by=/usr/share/keyrings/mongodb-server-6.0.gpg ] https://repo.mongodb.org/apt/ubuntu jammy/mongodb-org/6.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-6.0.list
        ```
      - Ubuntu 20.04 (Focal)
        ```

>         echo "deb [ arch=amd64,arm64 signed-by=/usr/share/keyrings/mongodb-server-6.0.gpg ] https://repo.mongodb.org/apt/ubuntu focal/mongodb-org/6.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-6.0.list

Reviewer 2: just call it `echo "deb [ arch=amd64,arm64 signed-by=/usr/share/keyrings/mongodb-server-6.0.gpg ] https://repo.mongodb.org/apt/ubuntu {version}/mongodb-org/6.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-6.0.list` and ask users to replace version based on their ubuntu maybe? It seems unnecessary to display same command for all four versions.

    ```

#### Mac Systems

> The following steps are for a Mac systems.

Reviewer: ```suggestion
The following steps are for a Mac system.
```

- Copy `.env.example` to `.env` and set all the environment variables as mentioned in `docs/ENV.md`. No need to change anything if you're planning to connect to a local DB. If you're planning to connect your local server to a staging and prod DB, only then you need to change. PLEASE DO NOT CONNECT YOUR LOCAL INSTANCE TO STAGING/PROD DB.

## Running locally

> for Linux Simply run the below in the terminal:
> ```
> ./startServerLinux.sh
> ```

Reviewer: ```suggestion
For Linux machines, simply run the below in the terminal:
```bash
chmod +x startServerLinux.sh
./startServerLinux.sh
```
```
